packages/medusa/src/services/cart.ts

```diff
@@ -1081,6 +1081,9 @@ class CartService extends TransactionBaseService {
           if (hasFreeShipping) {
             await this.adjustFreeShipping_(cart, true)
           }
+        } else if (isDefined(data.discounts) && !data.discounts.length) {
+          cart.discounts.length = 0
```

Comment:
here the .length will be 0 already

Reply:
On the cart? I don't think so, could you point me to the line that set it to an empty array before just to be sure?
In my opinion, you can have discounts on the cart, the client send an empty array, the cart must discounts must be presented. wdyt?

Reply:
oh man, disregard it... I read `data.discounts` not `cart` 🙄 
sorry haha

Reply:
no problem at all. You gave me a doubt for a second 😂